qa/rolling-upgrade/src/test/java/org/elasticsearch/upgrades/FieldCapsIT.java

```diff
@@ -0,0 +1,172 @@
+               "yellow_field": { "type": "long" },
+               "timestamp": {"type": "date"}
+             }
+            """;
+        if (CLUSTER_TYPE == ClusterType.OLD) {
+            createIndex("old_red_1", Settings.EMPTY, redMapping);
+            createIndex("old_red_2", Settings.EMPTY, redMapping);
+            createIndex("old_red_empty", Settings.EMPTY, redMapping);
+            createIndex("old_green_1", Settings.EMPTY, greenMapping);
+            createIndex("old_green_2", Settings.EMPTY, greenMapping);
+            createIndex("old_green_empty", Settings.EMPTY, greenMapping);
+            for (String index : List.of("old_red_1", "old_red_2", "old_green_1", "old_green_2")) {
+                final Request indexRequest = new Request("POST", "/" + index + "/" + "_doc/1");
+                indexRequest.addParameter("refresh", "true");
+                indexRequest.setJsonEntity(
+                    Strings.toString(JsonXContent.contentBuilder().startObject().field("timestamp", "2020-01-01").endObject())
+                );
+                assertOK(client().performRequest(indexRequest));
+            }
+        } else if (CLUSTER_TYPE == ClusterType.MIXED && FIRST_MIXED_ROUND) {
+            createIndex("new_red_1", Settings.EMPTY, redMapping);
+            createIndex("new_red_2", Settings.EMPTY, redMapping);
+            createIndex("new_red_empty", Settings.EMPTY, redMapping);
+            createIndex("new_green_1", Settings.EMPTY, greenMapping);
+            createIndex("new_green_2", Settings.EMPTY, greenMapping);
+            createIndex("new_green_empty", Settings.EMPTY, greenMapping);
+            for (String index : List.of("new_red_1", "new_red_2", "new_green_1", "new_green_2")) {
+                final Request indexRequest = new Request("POST", "/" + index + "/" + "_doc/1");
+                indexRequest.addParameter("refresh", "true");
+                indexRequest.setJsonEntity(
+                    Strings.toString(JsonXContent.contentBuilder().startObject().field("timestamp", "2020-10-10").endObject())
+                );
+                assertOK(client().performRequest(indexRequest));
+            }
+        }
+        final QueryBuilder indexFilter = QueryBuilders.rangeQuery("timestamp").gte("2020-01-01").lte("2020-12-12");
+
+        // old red indices without filter
+        {
+            FieldCapabilitiesResponse resp = fieldCaps(List.of("old_red_*"), List.of("*"), null);
+            assertThat(resp.getIndices(), equalTo(new String[] { "old_red_1", "old_red_2", "old_red_empty" }));
+            assertThat(resp.getField("red_field").keySet(), contains("keyword"));
+            assertTrue(resp.getField("red_field").get("keyword").isSearchable());
+            assertThat(resp.getField("yellow_field").keySet(), contains("integer"));
+            assertTrue(resp.getField("yellow_field").get("integer").isSearchable());
+        }
+        // old red indices with filter
```

Comment:
nit: given that we are introducing a filters parameter in field_caps, could you specify *index filter* in the comment? :)

Reply:
+1, fixed in https://github.com/elastic/elasticsearch/pull/84455/commits/b87784582f014709fc92395f21d9649ea09dd7ee